File: src/main/java/net/hectus/neobb/turn/legacy_game/warp/LTDefaultWarp.java
package net.hectus.neobb.turn.legacy_game.warp;

import net.hectus.neobb.turn.default_game.attributes.clazz.*;
import org.bukkit.World;

import java.util.List;

public class LTDefaultWarp extends LWarpTurn {
    public LTDefaultWarp(World world) { super(world, "legacy-default"); }

    @Override
    public int chance() {
        return 0;
    }

    @Override
    public List<Class<? extends Clazz>> allows() {
        return List.of(NeutralClazz.class, WaterClazz.class, NatureClazz.class, RedstoneClazz.class);
    }

    @Override
    public int cost() {
        return 0;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/legacy_game/warp/LTDefaultWarp.kt
package net.hectus.neobb.turn.legacy_game.warp

import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.turn.default_game.attribute.clazz.*
import kotlin.reflect.KClass

class LTDefaultWarp(data: PlacedStructure?, player: NeoPlayer?) : LWarpTurn(data, "default", player) {
    override val chance: Double = 0.0
    override val allows: List<KClass<out Clazz>> = listOf(NeutralClazz::class, WaterClazz::class, NatureClazz::class, RedstoneClazz::class)
    override val temperature: Temperature = Temperature.NORMAL
}